Libraries/TVector/TVector.h

```diff
@@ -0,0 +1,151 @@
+#ifndef TVector_h
+#define TVector_h
+#include "Tile.h"
+
+template <class T>
+class TVector {
+  private:
+    uint8_t iSize;
+    T *vector;
+
+  public:
+    TVector();//constructor with 1 element
+    TVector(const TVector &V);//constructor that copies another vector
+    ~TVector();
+
+    uint8_t getSize();
+
+    void pushTop(T t);
+    void pushFront(T t);
+    void popTop();
+    void popFront();
+    void set(T t, uint8_t index);
+    void erase(uint8_t index);
+
+    T& operator[](uint8_t index) const;
+    void operator=(const TVector &V);
+};
+
+template <class T>
+TVector<T>::TVector() {
+  iSize = 0;
+  vector = NULL;
+}
+
+template <class T>
+TVector<T>::TVector(const TVector &V) {
+  vector = new T[V.iSize];
+  
+  for(int i = 0 ; i < V.iSize ; i++) {
+    vector[i] = V[i];
+  }
+
+  iSize = V.iSize;
+}
+
+template <class T>
+TVector<T>::~TVector() {
+  delete[] vector;
+}
+
+template <class T>
+uint8_t TVector<T>::getSize() {
+  return iSize;
+}
+
+template <class T>
+void TVector<T>::pushTop(T t) {
+  T *newVector = new T[iSize + 1];
```

Comment:
Variable names should be like:
int number_of_books = 3;
float total_weight_of_books = 1.5;

In your case:
size is sufficient.